Server/controllers/leave.controller.js

```diff
@@ -1,18 +1,58 @@
 "use strict";
+const leavesDataLayer = require('../datalayer/leave.datalayer');
 /**
  * Class representing leaves controller
  *
  * @class leavesController
  */
 class leavesController {
 
-    static getLeavesByUserId (req, res) {}
+    static async getLeavesByUserId (req, res) {
+        const userId = req.query.user;
+        let leavesList;
+        try {
+            leavesList = await leavesDataLayer.getLeavesByUserId(userId);
+        } catch (err) {
+            console.log(err.message);
+            return res.status(400);
+        }
+        return res.status(200).json(leavesList);
+    }
 
-    static createLeave (req, res) {}
+    static async createLeave (req, res) {
+        let createdLeave;
+        try {
+            createdLeave = await leavesDataLayer.createLeave(req.body);
+        } catch(err) {
+            console.log(err.message);
+            return res.status(400);
+        }
+        return res.status(201).json(createdLeave);
+    }
 
-    static updateLeaveById (req, res) {}
+    static async updateLeaveById (req, res) {
+        const leaveId = req.params.leaveId;
+        let updatedLeave;
+        try {
+            updatedLeave = await leavesDataLayer.updateLeaveById(leaveId, req.body);
+        } catch(err) {
+            console.log(err.message);
+            return res.status(400);
+        }
+        return res.status(200).json(updatedLeave);
+    }
 
-    static deleteLeaveById (req, res) {}
+    static async deleteLeaveById (req, res) {
+        const leaveId = req.params.leaveId;
+        let deletedLeave;
+        try {
+            deletedLeave = await leavesDataLayer.deleteLeaveById(leaveId);
```

Comment:
Follow feedback of `getLeavesByUserId ` and `createLeave ` function.

Also what is returned in `deletedLeave` ? I believe client only want to know true or false in this case

Reply:
Found document is returned in case of deleteLeave. Changed response to boolean depending on document found.